services/ifrs/tests_ifrs/test_outbox_subscriber.py

```diff
@@ -0,0 +1,276 @@
+# Copyright 2021 - 2024 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for functionality related to the outbox subscriber."""
+
+from typing import Callable
+from unittest.mock import AsyncMock
+
+import pytest
+from ghga_event_schemas import pydantic_ as event_schemas
+from ghga_service_commons.utils.utc_dates import now_as_utc
+from hexkit.correlation import get_correlation_id
+from hexkit.custom_types import JsonObject
+from hexkit.providers.mongodb import MongoDbDaoFactory
+from ifrs.adapters.inbound import models
+from ifrs.adapters.inbound.utils import assert_record_is_new, make_record_from_update
+from ifrs.adapters.outbound.dao import (
+    get_file_deletion_requested_dao,
+)
+from logot import Logot, logged
+from pydantic import BaseModel
+
+from tests_ifrs.fixtures.joint import JointFixture
+
+CHANGE_EVENT_TYPE = "upserted"
+DELETE_EVENT_TYPE = "deleted"
+
+
+TEST_FILE_ID = "test_id"
+
+TEST_NONSTAGED_FILE_REQUESTED = event_schemas.NonStagedFileRequested(
+    file_id=TEST_FILE_ID,
+    target_object_id="",
+    target_bucket_id="",
+    s3_endpoint_alias="",
+    decrypted_sha256="",
+)
+
+TEST_FILE_UPLOAD_VALIDATION_SUCCESS = event_schemas.FileUploadValidationSuccess(
+    upload_date=now_as_utc().isoformat(),
+    file_id=TEST_FILE_ID,
+    object_id="",
+    bucket_id="",
+    s3_endpoint_alias="",
+    decrypted_size=0,
+    decryption_secret_id="",
+    content_offset=0,
+    encrypted_part_size=0,
+    encrypted_parts_md5=[],
+    encrypted_parts_sha256=[],
+    decrypted_sha256="",
+)
+
+TEST_FILE_DELETION_REQUESTED = event_schemas.FileDeletionRequested(file_id=TEST_FILE_ID)
+
+
+def test_make_record_from_update():
+    """Test the get_record function"""
+    record = make_record_from_update(
+        models.FileDeletionRequestedRecord, TEST_FILE_DELETION_REQUESTED
+    )
+    assert record.model_dump() == {
+        "correlation_id": get_correlation_id(),
+        "file_id": TEST_FILE_ID,
+    }
+
+
+@pytest.mark.asyncio()
```

Comment:
Use `pytestmark = pytest.mark.asyncio()` so you don't need to annotate it every time.

Reply:
Ok, then we should leave it like that.